sunpy/net/dataretriever/sources/adapt.py

```diff
@@ -0,0 +1,104 @@
+    ADAPTInputSource,
+    ADAPTLonType,
+    ADAPTMagData,
+    ADAPTRealizations,
+    ADAPTResolution,
+    ADAPTVersionMonth,
+    ADAPTVersionYear,
+)
+
+__all__ = ['ADAPTClient']
+
+
+class ADAPTClient(GenericClient):
+    """
+    Provides access to the ADvanced Adaptive Prediction Technique (ADAPT) products
+    of the National Solar Observatory (NSO).
+
+    `Searches data hosted by the NSO <https://gong.nso.edu/adapt/maps/gong/>`__
+
+    Example
+    --------
+    >>> # Imports
+    >>> import astropy.units as u
+    >>> from sunpy.net import Fido, attrs as a
+    >>> from sunpy.coordinates.sun import carrington_rotation_time
+    >>> from sunpy.net.dataretriever.attrs.adapt import ADAPTLonType
+
+    >>> # Define the Carrington Rotation Number and the number of frames
+    >>> CR = 2193
+    >>> frames = 10
+    >>> date_start = carrington_rotation_time(CR)
+    >>> date_end = date_start + frames*(3*1.9999999 * u.hour)
+
+    >>> # Format the Search Dates
+    >>> tstring = r"%Y-%m-%dT%H:%M:%S"
+    >>> get_date_start =date_start.strftime(tstring)
+    >>> get_date_end   =date_end.strftime(tstring)
+    >>> longitude_type = '0'
+
+    >>> res = Fido.search(a.Time(get_date_start, get_date_end), a.Instrument('adapt'), ADAPTLonType(longitude_type))
+
+    Output:
+
+    <class 'sunpy.net.fido_factory.UnifiedResponse'>
+    Results from 1 Provider:
+    <BLANKLINE>
+    1 Results from the ADAPTClient:
+    <BLANKLINE>
+            Start Time              End Time        Instrument Source Provider ADAPTFileType ... seconds_since_last_obs
+    ----------------------- ----------------------- ---------- ------ -------- ------------- ... ----------------------
+    2017-07-20 08:00:00.000 2017-07-20 08:00:59.999      ADAPT    NSO     GONG             4 ...                      0
+    <BLANKLINE>
+    <BLANKLINE>
+
+    References
+    ----------
+    `Names and possible attrs values are available <https://gong.nso.edu/adapt/maps/adapt_filename_notes.txt>`__.
+    """
+
+    baseurl = r'https://gong.nso.edu/adapt/maps/gong/%Y/adapt(\d){5}_(\d){2}(\w){1}(\d){3}_(\d){12}_(\w){1}(\d){8}(\w){1}(\d){1}\.fts\.gz'
+    pattern = '{}adapt{ADAPTFileType:1d}{ADAPTLonType:1d}{ADAPTInputSource:1d}{ADAPTDataAssimilation:1d}{ADAPTResolution:1d}' + \
+    '_{ADAPTVersionYear:2d}{ADAPTVersionMonth:1l}{ADAPTRealizations:3d}_{year:4d}{month:2d}{day:2d}{hour:2d}{minute:2d}' + \
+    '_{ADAPTEvolutionMode:1l}{days_since_last_obs:2d}{hours_since_last_obs:2d}{minutes_since_last_obs:2d}{seconds_since_last_obs:2d}{ADAPTHelioData:1l}{ADAPTMagData:1d}.fts.gz'
+
+    @classmethod
+    def _attrs_module(cls):
+        return 'adapt', 'sunpy.net.dataretriever.attrs.adapt'
+
+    @classmethod
+    def register_values(cls):
+        adict = {
+            a.Instrument: [('ADAPT', 'ADvanced Adaptive Prediction Technique.')],
+            a.Provider: [('NSO', 'National Solar Observatory.')],
+            a.Source: [('GONG', 'Global Oscillation Network Group.')],
```

Comment:
Is this correct?

Reply:
I believe so, though I wrote it a while back. Does something seem off?

Reply:
Ah I misread, ignore me.